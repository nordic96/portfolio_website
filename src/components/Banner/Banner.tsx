import React from 'react';
import LabelContainer from 'labelcontainer';

const Banner = () => {
    const lsInstance = LabelContainer.getInstance();
    return (
        <div
            className={
                'items-center relative gap-2 rounded-md flex lg:min-w-[70.5rem] max-sm:w-full py-2 px-2 justify-start bg-siablue text-white text-lg max-sm:text-sm text-start'
            }>
            <img
                className={
                    'absolute h-12 max-sm:h-8 -right-6 max-sm:-right-4 -top-8 max-sm:-top-6'
                }
                src={'assets/images/sailorcap.png'}
                alt={'sailorcap'}
            />
            <img
                src={'assets/icons/info_icon.png'}
                alt={'info_icon'}
                className={'h-6'}
            />
            {lsInstance.getLabel('banner_msg')}
        </div>
    );
};

export default Banner;
