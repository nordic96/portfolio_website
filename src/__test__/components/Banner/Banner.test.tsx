import React from 'react';
import Banner from '../../../components/Banner';
import { render, screen } from '@testing-library/react';
import LabelContainer from 'labelcontainer';

describe('Banner UI Component', () => {
    it('should render banner component', () => {
        render(<Banner />);
        expect(LabelContainer.getInstance).toHaveBeenCalled();
        expect(screen.getByText('mock_banner_msg')).toBeInTheDocument();
    });
});
