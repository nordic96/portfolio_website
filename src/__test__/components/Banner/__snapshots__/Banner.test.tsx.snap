// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner UI Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="items-center relative gap-2 rounded-md flex lg:min-w-[70.5rem] max-sm:w-full py-2 px-2 justify-start bg-siablue text-white text-lg max-sm:text-sm text-start"
      >
        <img
          alt="sailorcap"
          class="absolute h-12 max-sm:h-8 -right-6 max-sm:-right-4 -top-8 max-sm:-top-6"
          src="assets/images/sailorcap.png"
        />
        <img
          alt="info_icon"
          class="h-6"
          src="assets/icons/info_icon.png"
        />
        mock_banner_msg
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="items-center relative gap-2 rounded-md flex lg:min-w-[70.5rem] max-sm:w-full py-2 px-2 justify-start bg-siablue text-white text-lg max-sm:text-sm text-start"
    >
      <img
        alt="sailorcap"
        class="absolute h-12 max-sm:h-8 -right-6 max-sm:-right-4 -top-8 max-sm:-top-6"
        src="assets/images/sailorcap.png"
      />
      <img
        alt="info_icon"
        class="h-6"
        src="assets/icons/info_icon.png"
      />
      mock_banner_msg
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
